manifests/git_repo.pp

```diff
@@ -7,7 +7,7 @@
 
   validate_string($source)
 
-  $sources = {"${upstream_remote_name}" => "https://github.com/${source}.git"}
+  $sources = hash([$upstream_remote_name, "https://github.com/${source}.git"])
```

Comment:
I'm surprised this is needed. Maybe it's nicer to just define ```$sources``` where you define ```$sources_real```?